### ISSUE TITLE
Super amazing project!

### ISSUE THREAD

**Issue by oronsh** [vthriller/promql#6]
Hello,
I find your project so great because I haven't seen any `promql` parser  implementations other than golang.
Is this project production ready? Is it fully compatible with `prometheus' promql`?
Thanks a lot!

**Comment by JohnnyQQQQ** [comment]
I don't want to create a new issue so that I will piggyback on this one :) 

I'm currently writing a timeseries database that should be promQL compliant. This library is a good starting point for parsing the AST. I would love to help to get it more compliant and add missing features. While running the promQL compliance testsuite against my database I found a few things that would need to be added, like accepting numbers in their hexadecimal representation `0x3d` or accepting string literals `'a simple string'`.

@vthriller are you up for new contributors, or should I fork?

**Comment by vthriller** [comment]
@JohnnyQQQQ sorry for late response. I don't really know. I'm still interested in having up-to-date promql parser, but recent versions of nom are pretty finicky and hit recursion limit much earlier than before, and I'm not sure whether I want to keep going with, resort to earlier versions or just rewrite everying in pest/chomsky/pom/combine/whatever else there is these days. (At one point I tried rewriting it using lalrpop, and it didn't go well. Can't remember the details though.)

So in short, I guess you're better off forking, but PRs with smaller fixes are still welcome.
